Adding rich HTML output for All of GEOPHIRES.

malcolm-dsider opened the pull request:
New code for adding optional HTML output to all GEOPHIRES modules. It also offers an optional better text version of the output.

There is also one fix for the OUTPUT unit change function (but it doesn't solve the currency exchange issue).

Because this messes with the output format, the unit testing parsing routines are broken, so some of the unit tests fail. This creates a new issue to be resolved.

softwareengineerprogrammer commented on tests/examples/example_SHR-2.out:
```diff
@@ -4,235 +4,220 @@
-      Number of segments:                                     1
-      Geothermal gradient:                                    0.0500 degC/m
-
+      Economic Model                               : Fixed Charge Rate (FCR)
```
What is the motivation for aligning the colons like this instead of the standard/regular way they currently? I don't believe I've ever seen this style of textual representation for data.

malcolm-dsider commented:
It made it easier to align the numbers, that is all – I am essentially padding the text on either side and allowing Python to handle the alignment rather than hand-padding the alignment with spaces on a line-by-line basis for alignment. Doing this lets me write the report lines in a generic loop rather than a custom-line-by-line format. More compact.

 

From: Jonathan Pezzino ***@***.***> 
Sent: Thursday, April 4, 2024 8:52 AM
To: NREL/GEOPHIRES-X ***@***.***>
Cc: Malcolm Ross ***@***.***>; Author ***@***.***>
Subject: Re: [NREL/GEOPHIRES-X] Adding rich HTML output for All of GEOPHIRES. (PR #177)

 

@softwareengineerprogrammer commented on this pull request.

  _____  

In tests/examples/example_SHR-2.out <https://github.com/NREL/GEOPHIRES-X/pull/177#discussion_r1551734087> :

> -      Production Wellbore heat transmission calculated with Ramey's model
-      Average production well temperature drop:              19.0 degC
-      Flowrate per production well:                          55.0 kg/sec
-      Injection well casing ID:                               7.000 in
-      Production well casing ID:                              7.000 in
-      Number of times redrilling:                             0
-      Power plant type:                                       Single-Flash
-
-
-                         ***RESOURCE CHARACTERISTICS***
-
-      Maximum reservoir temperature:                        500.0 degC
-      Number of segments:                                     1
-      Geothermal gradient:                                    0.0500 degC/m
-
+      Economic Model                               : Fixed Charge Rate (FCR)

What is the motivation for aligning the colons like this instead of the standard/regular way they are currently? I don't believe I've ever seen this style of textual representation for data.

—
Reply to this email directly, view it on GitHub <https://github.com/NREL/GEOPHIRES-X/pull/177#pullrequestreview-1979987885> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWGVYT4D2BCXPU6ZOBE6IIDY3VLGJAVCNFSM6AAAAABFVWRC26VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNZZHE4DOOBYGU> .
You are receiving this because you authored the thread.  <https://github.com/notifications/beacon/AWGVYT4YBN2RN3ADKPBTTUTY3VLGJA5CNFSM6AAAAABFVWRC26WGG33NNVSW45C7OR4XAZNRKB2WY3CSMVYXKZLTORJGK5TJMV32UY3PNVWWK3TUL5UWJTTWAQ322.gif> Message ID: ***@***.*** ***@***.***> >

malcolm-dsider commented:
This PR now includes the code for PTC and ITC.
It also corrects some issues in the output unit test files

softwareengineerprogrammer commented:
@malcolm-dsider took a look at fixing [failing tests](https://github.com/NREL/GEOPHIRES-X/actions/runs/8637350288/job/23679321834?pr=177) per your request.

The [extended economic profile test](https://github.com/NREL/GEOPHIRES-X/blob/46d18a783f1134f9873e77899b39a5bb2d009817/tests/test_geophires_x_client.py#L200) is failing because extended economic profile is no longer present in `example1_addons.out`: https://github.com/NREL/GEOPHIRES-X/pull/177/files#diff-d5df12751cf72e6cd45c0019dedbb262986aa5d2b7754d70799836a8bb8cb027L247-L300.

Can you please advise on whether this was intentional? (if so, we need to discuss because it would be backwards-incompatible)

malcolm-dsider commented:
Yep, that was a bug I introduced in the outputput. That bug is now fixed and it is correctly adding the outputs report for addons. I did a commit and push on the fix. But the unit tests are still failing – it looks like a parser issue?

 

From: Jonathan Pezzino ***@***.***> 
Sent: Thursday, April 11, 2024 7:33 AM
To: NREL/GEOPHIRES-X ***@***.***>
Cc: Malcolm Ross ***@***.***>; Mention ***@***.***>
Subject: Re: [NREL/GEOPHIRES-X] Adding rich HTML output for All of GEOPHIRES. (PR #177)

 

@malcolm-dsider <https://github.com/malcolm-dsider>  took a look at fixing failing tests <https://github.com/NREL/GEOPHIRES-X/actions/runs/8637350288/job/23679321834?pr=177>  per your request.

The extended economic profile test <https://github.com/NREL/GEOPHIRES-X/blob/46d18a783f1134f9873e77899b39a5bb2d009817/tests/test_geophires_x_client.py#L200>  is failing because extended economic profile is no longer present in example1_addons.out: https://github.com/NREL/GEOPHIRES-X/pull/177/files#diff-d5df12751cf72e6cd45c0019dedbb262986aa5d2b7754d70799836a8bb8cb027L247-L300.

Can you please advise on whether this was intentional? (if so, we need to discuss because it would be backwards-incompatible)

—
Reply to this email directly, view it on GitHub <https://github.com/NREL/GEOPHIRES-X/pull/177#issuecomment-2049593334> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWGVYT3Y3XOUBLB52WQJYJ3Y4Z7GPAVCNFSM6AAAAABFVWRC26VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANBZGU4TGMZTGQ> .
You are receiving this because you were mentioned.  <https://github.com/notifications/beacon/AWGVYTYCRYZMHFGWNSYBROTY4Z7GPA5CNFSM6AAAAABFVWRC26WGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTT2FJH7M.gif> Message ID: ***@***.*** ***@***.***> >

softwareengineerprogrammer reviewed (approved):
Approved

malcolm-dsider commented on src/geophires_x_client/geophires_x_result.py:
```diff
@@ -140,7 +140,7 @@ class GeophiresXResult:
                 #  'Annual Thermal Drawdown (%/year)',
                 'Bottom-hole temperature',
-                'Well seperation: fracture height',  # TODO correct typo upstream
+                'Well separation: fracture height',
```
Yes, and the documentation spells it the right way ("separation"), so technically, this is a fix - if a user was using the parameter with the documented spelling, it was being ignored - fixing this means it will actually be functioning correctly!
